speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -161,3 +166,85 @@ def extract_features(self, wav):
             out = F.layer_norm(out, out.shape)
 
         return out
+
+
+class HuggingFaceWav2Vec2Pretrain(nn.Module):
+    """This lobe enables the integration of HuggingFace
+     wav2vec2.0 models to be pretrained.
+
+    Source paper: https://arxiv.org/abs/2006.11477
+    Transformer from HuggingFace needs to be installed:
+    https://huggingface.co/transformers/installation.html
+
+    The return is an HuggingFace format and the mask indices that contains:
+    https://huggingface.co/transformers/model_doc/wav2vec2.html#wav2vec2forpretraining
+
+    For instance, it returns the loss that can be accessed with .loss
+
+    Arguments
+    ---------
+    source : str
+        HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+    save_path : str
+        Path (dir) of the downloaded model.
+    mask_prob : float (default: 0.65)
+        Probability of masking a given frame. Default is taken from the paper.
+    mask_length : float (default: 10)
+        Length (i.e. number of consecutive masked frames). Default is taken from
+        the paper.
+    Example
+    -------
+    >>> inputs = torch.rand([10, 600])
+    >>> model_hub = "facebook/wav2vec2-base-960h"
+    >>> save_path = "savedir"
+    >>> model = HuggingFaceWav2Vec2Pretrain(model_hub, save_path)
+    >>> outputs = model(inputs)
+    >>> outputs.shape
+    torch.Size([10, 1,  768])
+    """
+
+    def __init__(
+        self, source, save_path, mask_prob=0.65, mask_length=10,
+    ):
+        super().__init__()
+
+        self.mask_prob = mask_prob
+        self.mask_length = mask_length
+
+        # Download the config of the model from HuggingFace.
+        config = Wav2Vec2Config.from_pretrained(source, cache_dir=save_path)
+        config.gradient_checkpointing = (
+            False  # This cause errors with DDP if True.
```

Comment:
@TParcollet , is it possible to add `run_on_main()` for that part for gradient checkpointing... ?

Reply:
Nope. This is within HuggingFace ... :-/ We must disable it.